fs/fstest/file.go

```diff
@@ -158,11 +159,15 @@ func Link(oldname, newname string) Applier {
 func CreateSocket(name string, perm os.FileMode) Applier {
 	return applyFn(func(root string) error {
 		fullPath := filepath.Join(root, name)
-		ln, err := net.Listen("unix", fullPath)
+		fd, err := unix.Socket(unix.AF_UNIX, unix.SOCK_STREAM, 0)
 		if err != nil {
 			return err
 		}
-		defer ln.Close()
+		defer unix.Close(fd)
+		sa := &unix.SockaddrUnix{Name: fullPath}
+		if err := unix.Bind(fd, sa); err != nil {
+			return err
+		}
 		return os.Chmod(fullPath, perm)
```

Comment:
If this fails you return an error but the socket is left on disk. I suspect this isn't a problem because the caller is using this to populate some sort of temp dir which it will clean up in its entirety on error, but something to be aware of I guess.

Reply:
I think it's fine, but I'll let @dmcgowan decide